packages/importer/src/columns.ts

```diff
@@ -149,23 +153,80 @@ export const coerceValue = (
         ? { value: parseMultiple(String(value)), isError: false }
         : { value: null, isError: true };
     }
+    case 'file': {
+      const res = await parseFile((value as string).trim(), proxyFunction);
+      if (!res) return { value: null, isError: true };
+      return {
+        value: { name: 'upload', mediaType: res.mediaType, base64Content: res.base64Content } as XataFile,
+        isError: false
+      };
+    }
+    case 'file[]': {
+      // Cannot import multiple raw Base64 files because we will split in the wrong place
+      const promises = (value as string).split(/[,;|]/).map((uri) => parseFile(uri, proxyFunction));
+      const files = await Promise.all(promises);
+      const isError = files.some((file) => file === null);
+      const formatted = files.map(
+        (file) => ({ name: 'upload', mediaType: file?.mediaType, base64Content: file?.base64Content } as XataFile)
+      );
+      return { value: compact(formatted), isError };
+    }
     default: {
       return { value: null, isError: true };
     }
   }
 };
 
-export const coerceRows = <T extends Record<string, unknown>>(
+const parseFile = async (url: string, proxyFunction: ColumnOptions['proxyFunction']): Promise<XataFile | null> => {
+  const uri = url.trim();
+  try {
+    if (uri.startsWith('data:')) {
+      const [mediaType, base64Content] = uri.split(',');
+      return new XataFile({ base64Content, mediaType });
+    } else if (uri.startsWith('http://') || uri.startsWith('https://')) {
+      const validUrl = new URL(uri);
+      if (proxyFunction) {
+        const response = await proxyFunction(validUrl.href);
+        if (!response) return null;
+        return XataFile.fromBase64(response.base64Content, { mediaType: response.mediaType });
```

Comment:
I suppose we could change the type here and have the proxyFunction return Blob, but I couldn't get it to work (and I think the mime type would be lost) so I switched the type to make the proxy return the base64 content.